sdk/python/kfp/components/pipeline_channel.py

```diff
@@ -170,8 +171,7 @@ class PipelineParameterChannel(PipelineChannel):
         constructs it directly (for providing an immediate value), or it is a
         pipeline function argument.
       pattern: The serialized string regex pattern this pipeline channel created
-        from.
-      value: The actual value of the pipeline channel. If provided, the
+      default_value: The actual value of the pipeline channel. If provided, the
```

Comment:
Pipeline channel is transient concept that exists during the pipeline compilation process. It's a handle to a value which may be realized or unrealized at compile time. It's different from an inputSpec/outputSpec concept, the latter also exists post-compilation.
One example where `value` make more sense: `with dsl.ParallelFor([1, 2, 3]) as item:`, we create a pipeline channel that holds value `[1, 2, 3]`. It's not default value because there's no way it would be replaced later. The pipeline channel is used during compilation for downstream steps to reference the value. Post compilation, the pipeline channel no longer exists.

Reply:
This makes sense and helped me understand the existing code a bit more. I see that the current code slightly overloads the usage of pipeline channel's `value` field, using it to surface defaults to the pipeline. With some (different) refactoring, this is not necessary and the code becomes much simpler.

I will implement this different change alongside the optional artifact inputs work, per your other suggestion.